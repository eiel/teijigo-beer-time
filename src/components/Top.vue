<template>
  <div class="top">
    <h1>{{ msg }}</h1>
    <div class="marquee">
      <div class="marquee-inner">
        M3-2018秋 参戦決定！！！！衝撃のデビューを見逃すな！！！！
      </div>
    </div>

  </div>
</template>

<script>
export default {
  name: 'Top',
  data() {
    return {
      msg: '定時後ビールタイム非公式サイト',
    };
  },
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
h1, h2 {
  font-weight: normal;
}

.marquee {
  overflow: hidden;
  position: relative;
  background-color: black;
  padding: 10px;
}

.marquee::after {
  content: "";
  white-space: nowrap;
  display: inline-block;
}

.marquee > .marquee-inner {
  position: absolute;
  top: 10;
  white-space: nowrap;
  animation-name: marquee;
  animation-timing-function: linear;
  animation-duration: 20s;
  animation-iteration-count: infinite;
  color: orange;
}

.marquee > .marquee-inner:hover {
  animation-play-state: paused;
  cursor: default;
}

@keyframes marquee {
    0% { left: 100%; transform: translate(0); }
  100% { left: 0; transform: translate(-100%); }
}
</style>
