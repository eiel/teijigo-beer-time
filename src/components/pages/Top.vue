<template>
  <default-template>
    <div class="top">
      <tbt-h1 class="blink">{{ msg }}</tbt-h1>
      <tbt-marquee>
        M3-2018秋 参戦予定！！！！衝撃のデビューを見逃すな！！！！
      </tbt-marquee>
      <tbt-access-counter-section v-bind="counter" />
      <tbt-circle-cut-section />
      <p>
        <router-link to="/disco">
          ＞＞＞＞DISCOGRAPHY＜＜＜＜
        </router-link>
      </p>
      <p>
        <router-link to="/bbs">
          ★★★★★一言掲示板★★★★★
        </router-link>
      </p>
      <p>
        <router-link to="/member">
          🍺🍺🍺🍺️メンバー紹介🍺🍺🍺🍺
        </router-link>
      </p>
    </div>
  </default-template>
</template>

<script lang="ts">
import Vue from 'vue';
import Component from 'vue-class-component';
import firebase from 'firebase';
import database from '@/firebase-config';
import DefaultTemplate from '@/components/templates/Default';
import TbtH1 from '@/components/atoms/TbtH1';
import TbtMarquee from '@/components/atoms/TbtMarquee';
import TbtAccessCounterSection from '@/components/organisms/TbtAccessCounterSection';
import TbtCircleCutSection from '@/components/organisms/TbtCircleCutSection';

@Component({
  components: {
    DefaultTemplate,
    TbtH1,
    TbtMarquee,
    TbtAccessCounterSection,
    TbtCircleCutSection,
  },
})
export default class Top extends Vue {
  msg: string = '定時後ビールタイム公式サイト';
  counter = { access: 0, realtime: 0 };

  created() {
    this.countUp();
    this.listen();
  }

  listen() {
    database
      .ref('access_counter')
      .on('value', (snapshot: firebase.database.DataSnapshot | null) => {
        if (snapshot) {
          const realtime = parseInt(snapshot.val(), 10);
          this.counter = { ...this.counter, realtime };
        }
      });
  }

  countUp() {
    database
      .ref('access_counter')
      .once('value')
      .then((snapshot: firebase.database.DataSnapshot | null) => {
        if (snapshot) {
          const access = parseInt(snapshot.val(), 10) + 1;
          database.ref().update({ access_counter: access });
          this.counter = { ...this.counter, access };
        }
      });
  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
.blink {
  animation: blink 1s step-end infinite;
  -webkit-animation: blink 1s step-end infinite;
}

@keyframes blink {
  75% {
    opacity: 0;
  }
}

@-webkit-keyframes blink {
  75% {
    opacity: 0;
  }
}
</style>
