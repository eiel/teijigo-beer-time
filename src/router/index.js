import Vue from 'vue';
import Router from 'vue-router';
import Top from '@/components/Top';
import Disco from '@/components/Disco';

Vue.use(Router);

export default new Router({
  routes: [
    {
      path: '/',
      name: 'Top',
      component: Top,
    },
    {
      path: '/disco',
      name: 'Disco',
      component: Disco,
    },
  ],
});
