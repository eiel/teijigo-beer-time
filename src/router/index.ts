import Vue from 'vue';
import Router from 'vue-router';
import Top from '@/components/Top.vue';
import Disco from '@/components/Disco.vue';
import NotFound from '@/components/NotFound.vue';

Vue.use(Router);

export default new Router({
  mode: 'history',
  routes: [
    {
      path: '/',
      name: 'Top',
      component: Top,
    },
    {
      path: '/disco',
      name: 'Disco',
      component: Disco,
    },
    {
      path: '*',
      name: 'NotFound',
      component: NotFound,
    },
  ],
});
