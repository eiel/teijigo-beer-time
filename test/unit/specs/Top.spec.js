import Vue from 'vue';
import Top from '@/components/Top';
import router from '@/router/index';

describe('Top.vue', () => {
  it('should render correct contents', () => {
    const Constructor = Vue.extend(Top);
    const vm = new Constructor({ router }).$mount();
    expect(vm.$el.querySelector('.top h1').textContent)
      .toEqual('定時後ビールタイム公式サイト');
  });
});