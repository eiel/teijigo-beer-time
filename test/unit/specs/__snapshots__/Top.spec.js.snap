// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Top.vue should match the snapshot 1`] = `
<div
  id="app"
>
  <nav
    class="menuContainer"
    id="menu-container"
  >
    <ul
      class="menu"
    >
      <li
        class="menuItem"
      >
        <a
          class="router-link-active"
          href="/"
        >
          
        トップ
      
        </a>
      </li>
       
      <li
        class="menuItem"
      >
        <a
          class=""
          href="/disco"
        >
          
        DISCOGRAPHY
      
        </a>
      </li>
       
      <li
        class="menuItem"
      >
        <a
          class=""
          href="/member"
        >
          
        メンバー紹介
      
        </a>
      </li>
       
      <li
        class="menuItem"
      >
        <a
          class=""
          href="/bbs"
        >
          
        一言掲示板
      
        </a>
      </li>
    </ul>
  </nav>
   
  <div
    class="mainContainer"
    id="main-container"
  >
    <a
      class="router-link-active"
      href="/"
    >
      <img
        class="logo"
        src="@/assets/logo_1.png"
      />
    </a>
     
    <div
      class="top"
    >
      <h1
        class="h1 blink"
      >
        定時後ビールタイム公式サイト
      </h1>
       
      <p
        class="blink"
      >
        🎉🎉🎉🎉🎉🎉🎉祝！M3当選！🎉🎉🎉🎉🎉🎉
      </p>
       
      <div
        class="marquee"
      >
        <div
          class="marqueeInner"
        >
          
      M3-2018秋 参戦決定！！！！衝撃のデビューを見逃すな！！！！
    
        </div>
      </div>
       
      <section
        class="section"
      >
        <h2
          class="h2"
        >
          アクセスカウンター
        </h2>
         
        <p>
          あなたは0人目の訪問者です。
        </p>
         
        <p>
          現在の累計訪問者数は0人です。
        </p>
         
        <small>
          <a
            class=""
            href="/bbs"
          >
            キリ番報告はこちらで！！！
          </a>
        </small>
      </section>
       
      <section
        class="section"
      >
        <h2
          class="h2"
        >
          サークルカット
        </h2>
         
        <img
          alt="サークルカット"
          class="circleCut"
          src="@/assets/cut.png"
        />
      </section>
       
      <p>
        <a
          class=""
          href="/disco"
        >
          
        ＞＞＞＞DISCOGRAPHY＜＜＜＜
      
        </a>
      </p>
       
      <p>
        <a
          class=""
          href="/bbs"
        >
          
        ★★★★★一言掲示板★★★★★
      
        </a>
      </p>
       
      <p>
        <a
          class=""
          href="/member"
        >
          
        🍺🍺🍺🍺️メンバー紹介🍺🍺🍺🍺
      
        </a>
      </p>
    </div>
     
    <footer
      class="footer"
    >
      <p>
        ©定時後ビールタイム
      </p>
    </footer>
  </div>
</div>
`;
